### PR TITLE
ci: Add CI/CD and semantic release

### PR DESCRIPTION
**Pull request by rmi22186**
## Instructions
 1. PR target branch should be against `development`
 2. PR title name should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-title-check.yml
 3. PR branch prefix should follow this format: https://github.com/mParticle/mparticle-workflows/blob/main/.github/workflows/pr-branch-check-name.yml

 ## Summary
Add semantic release and ci/cd scripts/config

 ## Testing Plan
 - [ ] Was this tested locally? If not, explain why.
It is not possible to test a github action prior to release. However, if there are any failures, we can debug once this file is in master.

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5985

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 0
+                  ref: master
+
+            - name: Import GPG Key
+              uses: crazy-max/ghaction-import-gpg@v4
+              with:
+                  gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+                  passphrase: ${{ secrets.GPG_PASSPHRASE }}
```
Comment (alexs-mparticle) [inline .github/workflows/release.yml:89]: Don't forget to migrate these secrets!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                  git_user_signingkey: true
+                  git_commit_gpgsign: true
+
+            - name: Merge release branch into master branch
+              run: |
+                  git pull origin release/${{ github.run_number }}
+            - name: Setup Node.js
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16.x
```
Comment (alexs-mparticle) [inline .github/workflows/release.yml]: Probably should update this to something like v18 or v20

```suggestion
                  node-version: 20.x
```

**Review by alexs-mparticle — APPROVED** [review]
Looks good. Only one little reminder about secrets.

**Review by alexs-mparticle — APPROVED** [review]
Looks good. Only one little reminder about secrets.

**Comment by rmi22186** [comment]
@alexs-mparticle - yup i added the secrets yesterday also!

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        runs-on: ubuntu-latest
+        needs: confirm-public-repo-master-branch
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v3
+
+            - name: NPM install
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 18.x
```
Comment (rmi22186) [inline .github/workflows/release.yml]:
```suggestion
                  node-version: 20.x
```

**Pull request merged**